pkg/agent/controller/egress/egress_controller.go

```diff
@@ -388,6 +406,17 @@ func (c *EgressController) Run(stopCh <-chan struct{}) {
 	klog.Infof("Starting %s", controllerName)
 	defer klog.Infof("Shutting down %s", controllerName)
 
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(0)
+	eventBroadcaster.StartRecordingToSink(&v1.EventSinkImpl{
+		Interface: c.k8sClient.CoreV1().Events(""),
+	})
+	c.record = eventBroadcaster.NewRecorder(
+		scheme.Scheme,
+		corev1.EventSource{Component: controllerName},
+	)
```

Comment:
I'm confused by the code, why does it override the already initiailized members with exactly same code?

Reply:
because the instance of eventBroadcaster where we start logging is different, if we do not override it the events won't be logged. So, for this there are two approaches:
1) Overriding in the Run function as is done above.
2) We add eventBroadcaster to the structure of egressController, so that we can startLogging for the same instance.

@tnqn which approach should i follow?

Reply:
2 is the right way

Reply:
okay